web_widget_text_markdown/controllers/main.py

```diff
@@ -0,0 +1,32 @@
+import base64
+import unicodedata
+import logging
+from odoo import http
+from odoo.http import request
+_logger = logging.getLogger(__name__)
+
+
+class MarkdownController(http.Controller):
+    # This is a modified method from web>controllers>main 'upload_attachment'
+    @http.route(
+        '/web_widget_text_markdown/upload_attachment',
+        type='http', auth="user")
+    def upload_markdown_attachment(self, model, res_id, ufile):
+        attachment_obj = request.env['ir.attachment']
+        filename = unicodedata.normalize('NFD', ufile.filename) \
+            if request.httprequest.user_agent.browser == 'safari' \
+            else ufile.filename
+
+        try:
+            attachment = attachment_obj.create({
+                'name': filename,
+                'datas': base64.encodestring(ufile.read()),
+                'datas_fname': filename,
+                'res_model': model,
+                'res_id': int(res_id)
+            })
+        except Exception:
```

Comment:
can we track any specific exception?

Reply:
Removed method... now uses the Odoo 'original' (looks a bit hackish, but not repeated code for the same purpose)